src/PageHeader/PageHeader.examples.stories.tsx

```diff
@@ -136,6 +143,65 @@ export const PullRequestPage = () => (
   </Box>
 )
 
+export const IssuesPage = () => (
```

Comment:
This is an example to show how the focus order breaks when the heading has an interactive element inside. I plan to remove this story before merging. It is here for the code review. 

Note: I'll address warning (or even throwing an error with invariant) the cases where the heading includes an interactive element in a follow up PR.